Purchases/Purchasing/ProductsManager.swift

```diff
@@ -41,14 +41,17 @@ class ProductsManager: NSObject {
            self.systemInfo.useStoreKit2IfAvailable {
             _ = Task<Void, Never> {
                 do {
-                    let storeProduct = try await self.sk2StoreProduct(withIdentifiers: identifiers)
-                    completion(.success(storeProduct))
+                    let products = try await self.sk2StoreProducts(withIdentifiers: identifiers)
+                        .map(StoreProduct.from(product:))
+                    completion(.success(Set(products)))
                 } catch {
                     completion(.failure(error))
                 }
             }
         } else {
-            productsFetcherSK1.products(withIdentifiers: identifiers, completion: completion)
+            productsFetcherSK1.products(withIdentifiers: identifiers) { result in
+                completion(result.map { Set($0.map(StoreProduct.from(product:))) })
```

Comment:
This is just a Swift question from an android engineer.. what is going on here? what are we passing to `StoreProduct.from`?

Reply:
This is syntactic sugar, `$0.map` accepts a function that takes a single object, so `StoreProduct.from(product:)` is the function that is being passed.
It's the same as:

```swift
tacos.map { taco in 
  return StoreProduct.from(product: taco)
}
```